File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/base/AbstractEntityService.java
package com.espindola.lobwebapp.service.base;

import java.util.List;

import com.espindola.lobwebapp.domain.base.AbstractEntity;
import com.espindola.lobwebapp.repository.contract.AbstractEntityRepository;
import com.espindola.lobwebapp.service.contract.EntityService;

public abstract class AbstractEntityService<TEntity extends AbstractEntity>
		implements EntityService<TEntity> {

	private AbstractEntityRepository<TEntity> repository;

	public AbstractEntityService(AbstractEntityRepository<TEntity> repository) {
		this.repository = repository;

	}

	@Override
	public TEntity get(TEntity entity) {
		return repository.findOne(entity.getId());
	}

	@Override
	public TEntity find(Long id) {
		return repository.findOne(id);
	}

	@Override
	public List<TEntity> list() {
		return repository.findAll();
	}

	@Override
	// @Transactional Spring JPA is already annotated with this
	public void save(TEntity entity) {
		repository.save(entity);
	}

	@Override
	public void update(TEntity entity) {
		repository.save(entity);
	}

	@Override
	public void delete(Long id) {
		repository.delete(repository.findOne(id));
	}

}

File: lobwebapp-html/src/main/service/impl/base/AbstractEntityService.ts
///<reference path='../../../../../ts-definitions/DefinitelyTyped/angularjs/angular.d.ts'/>
///<reference path='../../../domain/base/AbstractEntity.ts'/>
///<reference path='../../contract/base/EntityService.ts'/>

module service.impl.base{
    export class AbstractEntityService<T extends domain.base.AbstractEntity> implements service.contract.base.EntityService<T> {
        
        private rootUrl : string = "http://localhost:8080/lobwebapp-core/rest";
        private http : ng.IHttpService;
        
        constructor(contextUrl: string, $http: ng.IHttpService){
            this.rootUrl += '/' + contextUrl;
            this.http = $http;
        }
        
        public save (entity: T, 
                successCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any, 
                errorCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig)=> any)
                {
                    this.getHttp().post(this.rootUrl, entity).success(successCallback).error(errorCallback);
                }
                
        public update (entity: T, 
                successCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any, 
                errorCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig)=> any) 
                {
                    this.getHttp().put(this.rootUrl, entity).success(successCallback).error(errorCallback);
                }
             
        public remove (entity: T, 
                successCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any, 
                errorCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig)=> any)
                {
                    this.getHttp().delete(this.rootUrl, entity).success(successCallback).error(errorCallback);
                }
                
        public findById (id : number,
                successCallback: (data: T, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any, 
                errorCallback: (data: T, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig)=> any)
                {
                    this.getHttp().get(this.rootUrl + '/'+ id).success(successCallback).error(errorCallback);
                }
                
        public list (successCallback: (data: T[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any, 
                errorCallback: (data: T[], status: number, headers: (headerName: string) => string, config: ng.IRequestConfig)=> any) 
                {
                    this.getHttp().get(this.rootUrl).success(successCallback).error(errorCallback);   
                }
        
        
        public getHttp() {
            return this.http;
        }
       
    }
}   